File: rcc/clients/dbus/upower_client.py
from rcc.clients.dbus.base.abstract_dbus_client import AbstractDbusClient
from rcc.utils.singleton import singleton


@singleton
class UpowerClient(AbstractDbusClient):
    """Dbus upower client"""

    def __init__(self):
        super().__init__(
            True,
            "org.freedesktop.UPower",
            "/org/freedesktop/UPower",
            "org.freedesktop.UPower",
        )

    @property
    def on_battery(self) -> bool:
        """Battery flag"""
        return self._get_property("OnBattery")


upower_client = UpowerClient()

File: RogPerfTuner/src/clients/dbus/linux/upower_client.cpp
#include "../../../../include/clients/dbus/linux/upower_client.hpp"

bool UPowerClient::isOnBattery() {
	return this->getProperty<bool>(QString("OnBattery"));
}

void UPowerClient::onBatteryChange(CallbackWithParams&& callback) {
	this->onPropertyChange("OnBattery", std::move(callback));
}

UPowerClient::UPowerClient()
	: AbstractDbusClient(true, QString("org.freedesktop.UPower"), QString("/org/freedesktop/UPower"), QString("org.freedesktop.UPower"), true) {
}